.github/workflows/lint.yml

```diff
@@ -40,4 +29,44 @@ jobs:
         run: |
           rustup update --no-self-update nightly
           rustup +nightly component add clippy
-          make clippy
\ No newline at end of file
+          make clippy
+
+  doc:
+    name: doc stable on ubuntu-latest
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@main
+      - name: Build docs
+        run: |
+          rustup update --no-self-update
+          make doc
+
+  version:
+    name: check rust version consistency
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@main
+        with:
+          profile: minimal
+          override: true
+      - name: check rust versions
+        run: ./scripts/check-rust-version.sh
+
+  changelog:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@main
+        with:
+          fetch-depth: 0
+      - name: Check if CHANGELOG.md is modified
+        run: |
+          # Get the list of changed files in the PR
+          changed_files=$(git diff --name-only origin/${{ github.event.pull_request.base.ref }}...${{ github.sha }})
+          # Check if CHANGELOG.md is in the list of changed files
+          if echo "$changed_files" | grep -q '^CHANGELOG.md$'; then
+            echo "CHANGELOG.md has been modified."
+          else
+            echo $'::warning file=CHANGELOG.md::CHANGELOG.md has not been modified.\n This warning can be ignored if is has been explicitely decided not to log changes.\n Except in this situation, make sure to add log changes.'
+            exit 1
```

Comment:
A downside to this is that this marks the overall CI run as a failure; somewhat destroying the glance value. If we think it will be rare to have a no-op changelog PR then this is probably fine.

Did you look into existing github actions we could use? e.g. this [one](https://github.com/tarides/changelog-check-action) allows you to skip the check by adding a `no changelog` label to the PR. I haven't actually audited the action itself though.

Reply:
I'm a bit hesitant to use actions that are not very popular - but skipping changelog check by adding `no changelog` label seems cool. Is this something that can be fairly easily emulated locally?

Reply:
Fair enough; I believe we should be able to simply copy the actions code. Its very short, label is added as an env variable and then simply used in an `if then` within the script.

```
env:
    NO_CHANGELOG_LABEL: ${{ contains(github.event.pull_request.labels.*.name, 'no changelog') }}
```

Reply:
In this PR I inspired myself with the action code and merged it with our existing one, enabling us to get the `no changelog` label feature: https://github.com/0xPolygonMiden/miden-vm/pull/1406

And  not having to depend on not widely used actions as @bobbinth mentioned